app/src/main/java/com/amaze/filemanager/utils/SmbUtil.java

```diff
@@ -97,4 +108,21 @@ public static SmbFile create(String path) throws MalformedURLException {
             .withCredentials(
                 new NtlmPasswordAuthentication(SingletonContext.getInstance(), uri.getUserInfo())));
   }
+
+  /**
+   * SMB version of {@link MainActivityHelper#checkFolder(File, Context)}.
+   *
+   * @param path SMB path
+   * @return {@link MainActivityHelper#DOESNT_EXIST} if specified SMB path doesn't exist on server,
+   *     else -1
+   */
+  public static int checkFolder(@NonNull String path) {
+    try {
+      SmbFile smbFile = create(path);
+      if (!smbFile.exists() || !smbFile.isDirectory()) return MainActivityHelper.DOESNT_EXIST;
+    } catch (SmbException | MalformedURLException e) {
+      Log.e(TAG, "Error checking folder existence", e);
+    }
+    return -1;
```

Comment:
Why return -1? Why not `MainActivityHelper.CAN_CREATE_FILES`?

Reply:
Good choice, but it's not over yet. When test on Fairphone 3 running LineageOS 16.0 (9.0) when returning `CAN_CREATE_FILES` files will not be overwritten - no actions at all; when return -1 seems will be overwritten without warning or dialogs.

Seems to be safest answer, but it's just not over yet. :/